File: MiniDOMBase/DMD/Anagrafica/Manutenzioni/Voci/VociManutenzioneCursor.cs
using System;
using DMD;
using DMD.Databases;
using minidom.repositories;
using static minidom.Sistema;

namespace minidom
{
    public partial class Anagrafica
    {


        /// <summary>
        /// Cursore sulla tabella delle voci di manutenzione
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class VociManutenzioneCursor 
            : minidom.Databases.DBObjectCursorPO<VoceManutenzione>
        {
            private DBCursorField<int> m_IDManutenzione = new DBCursorField<int>("IDManutenzione");
            private DBCursorStringField m_Categoria1 = new DBCursorStringField("Categoria1");
            private DBCursorStringField m_Descrizione = new DBCursorStringField("Descrizione");
            private DBCursorField<int> m_IDOggettoRimosso = new DBCursorField<int>("IDOggettoRimosso");
            private DBCursorStringField m_NomeOggettoRimosso = new DBCursorStringField("NomeOggettoRimosso");
            private DBCursorField<int> m_IDOggetto = new DBCursorField<int>("IDOggetto");
            private DBCursorStringField m_NomeOggetto = new DBCursorStringField("NomeOggetto");
            private DBCursorField<decimal> m_ValoreImponibile = new DBCursorField<decimal>("ValoreImponibile");
            private DBCursorField<decimal> m_ValoreIvato = new DBCursorField<decimal>("ValoreIvato");
            private DBCursorField<AzioneManutenzione> m_Azione = new DBCursorField<AzioneManutenzione>("Azione");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");

            /// <summary>
            /// Costruttore
            /// </summary>
            public VociManutenzioneCursor()
            {
            }

            /// <summary>
            /// Azione
            /// </summary>
            public DBCursorField<AzioneManutenzione> Azione
            {
                get
                {
                    return m_Azione;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<int> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            /// <summary>
            /// IDManutenzione
            /// </summary>
            public DBCursorField<int> IDManutenzione
            {
                get
                {
                    return m_IDManutenzione;
                }
            }

            /// <summary>
            /// Categoria1
            /// </summary>
            public DBCursorStringField Categoria1
            {
                get
                {
                    return m_Categoria1;
                }
            }

            /// <summary>
            /// Descrizione
            /// </summary>
            public DBCursorStringField Descrizione
            {
                get
                {
                    return m_Descrizione;
                }
            }

            /// <summary>
            /// IDOggettoRimosso
            /// </summary>
            public DBCursorField<int> IDOggettoRimosso
            {
                get
                {
                    return m_IDOggettoRimosso;
                }
            }

            /// <summary>
            /// NomeOggettoRimosso
            /// </summary>
            public DBCursorStringField NomeOggettoRimosso
            {
                get
                {
                    return m_NomeOggettoRimosso;
                }
            }

            /// <summary>
            /// IDOggetto
            /// </summary>
            public DBCursorField<int> IDOggetto
            {
                get
                {
                    return m_IDOggetto;
                }
            }

            /// <summary>
            /// NomeOggetto
            /// </summary>
            public DBCursorStringField NomeOggetto
            {
                get
                {
                    return m_NomeOggetto;
                }
            }

            /// <summary>
            /// ValoreImponibile
            /// </summary>
            public DBCursorField<decimal> ValoreImponibile
            {
                get
                {
                    return m_ValoreImponibile;
                }
            }

            /// <summary>
            /// ValoreIvato
            /// </summary>
            public DBCursorField<decimal> ValoreIvato
            {
                get
                {
                    return m_ValoreIvato;
                }
            }

            //public override object InstantiateNew(DBReader dbRis)
            //{
            //    return new VoceManutenzione();
            //}

            //public override string GetTableName()
            //{
            //    return "tbl_ManutenzioniVoci";
            //}

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Anagrafica.Manutenzioni.Voci; //.Module;
            }

            //protected internal override CDBConnection GetConnection()
            //{
            //    return APPConn;
            //}
        }
    }
}

File: MiniDOMBase/DMD/Anagrafica/Manutenzioni/Voci/VociManutenzioneCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica


    ''' <summary>
    ''' Cursore sulla tabella delle voci di manutenzione
    ''' </summary>
    ''' <remarks></remarks>
    Public Class VociManutenzioneCursor
        Inherits DBObjectCursorPO(Of VoceManutenzione)

        Private m_IDManutenzione As New CCursorField(Of Integer)("IDManutenzione")
        Private m_Categoria1 As New CCursorFieldObj(Of String)("Categoria1")
        Private m_Descrizione As New CCursorFieldObj(Of String)("Descrizione")
        Private m_IDOggettoRimosso As New CCursorField(Of Integer)("IDOggettoRimosso")
        Private m_NomeOggettoRimosso As New CCursorFieldObj(Of String)("NomeOggettoRimosso")
        Private m_IDOggetto As New CCursorField(Of Integer)("IDOggetto")
        Private m_NomeOggetto As New CCursorFieldObj(Of String)("NomeOggetto")
        Private m_ValoreImponibile As New CCursorField(Of Decimal)("ValoreImponibile")
        Private m_ValoreIvato As New CCursorField(Of Decimal)("ValoreIvato")
        Private m_Azione As New CCursorField(Of AzioneManutenzione)("Azione")
        Private m_Flags As New CCursorField(Of Integer)("Flags")


        Public Sub New()
        End Sub

        Public ReadOnly Property Azione As CCursorField(Of AzioneManutenzione)
            Get
                Return Me.m_Azione
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property IDManutenzione As CCursorField(Of Integer)
            Get
                Return Me.m_IDManutenzione
            End Get
        End Property

        Public ReadOnly Property Categoria1 As CCursorFieldObj(Of String)
            Get
                Return Me.m_Categoria1
            End Get
        End Property

        Public ReadOnly Property Descrizione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Descrizione
            End Get
        End Property

        Public ReadOnly Property IDOggettoRimosso As CCursorField(Of Integer)
            Get
                Return Me.m_IDOggettoRimosso
            End Get
        End Property

        Public ReadOnly Property NomeOggettoRimosso As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeOggettoRimosso
            End Get
        End Property

        Public ReadOnly Property IDOggetto As CCursorField(Of Integer)
            Get
                Return Me.m_IDOggetto
            End Get
        End Property

        Public ReadOnly Property NomeOggetto As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeOggetto
            End Get
        End Property

        Public ReadOnly Property ValoreImponibile As CCursorField(Of Decimal)
            Get
                Return Me.m_ValoreImponibile
            End Get
        End Property

        Public ReadOnly Property ValoreIvato As CCursorField(Of Decimal)
            Get
                Return Me.m_ValoreIvato
            End Get
        End Property

        Public Overrides Function InstantiateNew(dbRis As DBReader) As Object
            Return New VoceManutenzione
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_ManutenzioniVoci"
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Anagrafica.Manutenzioni.Voci.Module
        End Function

        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return APPConn
        End Function



    End Class


End Class